packages/preferences/src/node/json-preference-server.ts

```diff
@@ -86,8 +88,12 @@ export class JsonPreferenceServer implements PreferenceServer {
                 }
                 return this.fileSystem.resolveContent(uri).then(({ stat, content }) => {
                     const strippedContent = jsoncparser.stripComments(content);
-                    return JSON.parse(strippedContent)
-                })
+                    if (this.validator.validateJson(strippedContent)) {
```

Comment:
- validation should happen per a preference, not for the whole file
- it could be that we should parse manually the top-level object if `JSON.parse` cannot handle bogus JSON gracefully
- note that `ajv` is stateful, you should create a new instance per a file, or use `ajv.validate(schema, data)`